src/common/KokkosKernels_HashmapAccumulator.hpp

```diff
@@ -729,15 +811,63 @@ struct HashmapAccumulator
       hash_nexts[my_write_index] = hash_begins[hash];
       #endif
 
-      size_type hashbeginning = Kokkos::atomic_exchange(hash_begins+hash, my_write_index);
+      hashbeginning = Kokkos::atomic_exchange(hash_begins+hash, my_write_index);
       if (hashbeginning == -1) {
         used_hashes[Kokkos::atomic_fetch_add(used_hash_size, size_type(1))] = hash;
       }
       hash_nexts[my_write_index] = hashbeginning;
-      return INSERT_SUCCESS;
+      return __insert_success;
     }
   }
+  // end public members
+  private:
+    size_type __max_value_size;
+    size_type __hashOpRHS;
+    const int __insert_success;
+    const int __insert_full;
+
+    template<typename U = hash_type, typename std::enable_if<std::is_same<U, HashOpType::bitwiseAnd>::value, std::size_t>::type = 0>
+    KOKKOS_INLINE_FUNCTION
+    int __compute_hash(size_type key, size_type bitmask) {
+      //std::cout << "bitwiseAnd" << std::endl;
+      size_type hash = key & bitmask;
+      #ifdef HASHMAPACCUMULATOR_ASSERT_ENABLED
```

Comment:
@brian-kelley: I want to double check on this with you. `Kokkos::abort` runs on the GPU and generates a core file. It could be useful for debugging to leave these asserts in. Should we remove the debug blocks?

Reply:
@e10harvey Ah, forget I said anything :)
Didn't know abort could be called from device.